src/Command/ImportExisting.php

```
@@ -0,0 +1,309 @@
+	 *
+	 * @return void
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$statement = $this->old->executeQuery( "SELECT * FROM centrallog WHERE test = 'N'" );
+		$statement = $this->old->createQueryBuilder()
+			->select( 'cl.*, dmca.*, cp.*, u.*' )
+			->from( 'centrallog', 'cl' )
+			->leftJoin( 'cl', 'dmcatakedowns', 'dmca', 'cl.id = dmca.log_id' )
+			->leftJoin( 'cl', 'ncmecrelease', 'cp', 'cl.id = cp.log_id' )
+			->leftJoin( 'cl', 'user', 'u', 'cl.user = u.user' )
+			->where( "cl.test = 'N'" )
+			->andWhere( "cl.type IN ('Child Protection', 'DMCA')" )
+			->execute();
+
+		$promises = [];
+
+		$em = $this->doctrine->getEntityManager();
+
+		$result = $statement->fetchAll( \PDO::FETCH_NAMED );
+		foreach ( $result as $item ) {
+			// Get the username.
+			if ( !$item['wiki_user'] ) {
+				$item['wiki_user'] = self::USER_NAMES[$item['user'][0]];
+			}
+			$username = $item['wiki_user'];
+
+			$takedown = new Takedown( [
+				'id' => (int)$item['id'][0],
+				'reporter' => [
+					'username' => $username,
+				],
+				'created' => new \DateTime( $item['timestamp'][0] ),
+			] );
+
+			// Get the Page Ids.
+			$pageIds = [];
+			$siteId = null;
+			if ( $item['files_affected'] ) {
+				$urls = unserialize( $item['files_affected'] );
+				if ( $urls ) {
+					$pageIds = array_map( function ( $url ) {
+						$pieces = explode( '/', parse_url( $url,  PHP_URL_PATH ) );
+						return str_replace( ' ', '_', end( $pieces ) );
+					}, $urls );
+
+					$domain = parse_url( $urls[0],  PHP_URL_HOST );
+					$siteId = self::SITE_IDS[$domain];
+					$takedown->setSiteId( $siteId );
+				}
+			}
+
+			if ( $item['project'] ) {
+				$takedown->setSiteId( self::PROJECT_IDS[$item['project']] );
+			}
+
+			$takedown->setPageIds( $pageIds );
+
+			// Get Involved Users.
+			$involvedNames = [];
+			if ( $item['involved_user'] ) {
+				// @codingStandardsIgnoreLine
+				$involvedNames = @unserialize( $item['involved_user'] );
```

I'm not a fan of suppressing error messages, if you control the input it should be validated there that it is either an empty string, null, or a serialized object. And there are places where you do and you don't use '@' for unserialize

---

I completely agree. I did this because in the old database there is either serialized string, or usernames. There's not really a good way to tell if it's serialized or not without just running the serializer and suppressing errors.
https://stackoverflow.com/questions/1369936/check-to-see-if-a-string-is-serialized

Nothing in the new database is serialized, so we wont have to worry about this in the future. :)